### PR TITLE
Couchbase persistence recipe

### PR DESCRIPTION
**Pull request by johanandren**
Continuation on @ygree s work in #56

### CODE REVIEW

**Comment by johanandren** [comment]
I think what is missing is this part: 
https://github.com/lagom/lagom-recipes/pull/59/files#diff-ec4623c0086db747c4e35c01562828deR12

File: couchbase-persistence/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+Couchbase Lagom Samples
+=======================
+
+These are simple examples demonstrating how to use Couchbase with Lagom both in Java with Maven and in Scala with Sbt.
+These sample apps use PersistentEntity with Couchbase as a backend. Also they implement simple read-side processors that 
+consume events and update a couchbase document that can be queried.
+
+> Note: Couchbase specific parts are marked with `#couchbase-begin` and `#couchbase-end` in multiple fines.
```
Comment (ignasi35) [inline couchbase-persistence/README.md]:
```suggestion
> Note: Couchbase specific parts are marked with `#couchbase-begin` and `#couchbase-end` in multiple files.
```

File: couchbase-persistence/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+How to run
+----------
+
+In order to run these samples you'll need a Couchbase service. It also requires to pre-create necessary indexes.
+Docker image can be used to run Couchbase locally without installation.
+
+The Docker Compose preconfigured settings are available at 
+[Akka Persistence Couchbase GitHub](https://github.com/akka/akka-persistence-couchbase/tree/master/docker).
+
+Please see [Akka Persistence Couchbase Documentation](https://doc.akka.io/docs/akka-persistence-coucbase/current) for
```
Comment (ignasi35) [inline couchbase-persistence/README.md]:
```suggestion
Please see [Akka Persistence Couchbase Documentation](https://doc.akka.io/docs/akka-persistence-couchbase/current) for
```

**Comment by ignasi35** [comment]
Prefer `couchbase-persistence-java-mvn` and `couchbase-persistence-scala-sbt` for the folder names of each recipe.

File: couchbase-persistence/hello-service-scala/hello-impl/src/main/scala/org/example/hello/impl/HelloServiceImpl.scala [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  override def useGreeting(id: String) = ServiceCall { request =>
+    // Look up the hello entity for the given ID.
+    val ref = persistentEntityRegistry.refFor[HelloEntity](id)
+
+    // Tell the entity to use the greeting message specified.
+    ref.ask(UseGreetingMessage(request.message))
+  }
+
+  //#couchbase-begin
+  override def userGreetings(): ServiceCall[NotUsed, Seq[UserGreeting]] =
```
Comment (ignasi35) [inline couchbase-persistence/hello-service-scala/hello-impl/src/main/scala/org/example/hello/impl/HelloServiceImpl.scala]: I think we should add a comment here:

```
// This method demonstrates direct access to the DB using the `couchbaseSession` instance injected.
```

**Comment by johanandren** [comment]
Would be ok to merge this and that someone from the Lagom team picked it up? I think that would be much faster for the repository-best-practice changes.

File: couchbase-persistence/couchbase-persistence-scala-sbt/hello-impl/src/test/scala/org/example/hello/impl/HelloEntitySpec.scala [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  "hello entity" should {
+
+    "say hello by default" in withTestDriver { driver =>
+      val outcome = driver.run(Hello("Alice"))
+      outcome.replies should contain only "Hello, Alice!"
+    }
+
+    "allow updating the greeting message" in withTestDriver { driver =>
+      val outcome1 = driver.run(UseGreetingMessage("Hi"))
+      outcome1.events should contain only GreetingMessageChanged("Alice", "Hi")
```
Comment (ignasi35) [inline couchbase-persistence/couchbase-persistence-scala-sbt/hello-impl/src/test/scala/org/example/hello/impl/HelloEntitySpec.scala]: This assertion fails. `Alice` is not the  `id` of the entity instance, it's `hello-1` as set up in line `26`. The contents and this assertion are customized in this recipe (not part of the g8 template) which is why this test passes by default and not here.
```suggestion
      outcome1.events should contain only GreetingMessageChanged("hello-1", "Hi")
```

**Comment by ignasi35** [comment]
Sure! Once travis is green I'll squash-merge it and pick it up.

**Comment by johanandren** [comment]
Excellent, thanks!

**Pull request merged**